lib/portage/dbapi/bintree.py

```
@@ -199,25 +211,41 @@ def aux_update(self, cpv, values):
 				cpv = self._instance_key(cpv, support_string=True)[0]
 				build_id = cpv.build_id
 
-		tbz2path = self.bintree.getname(cpv)
-		if not os.path.exists(tbz2path):
+		binpkg_path = self.bintree.getname(cpv)
+		if not os.path.exists(binpkg_path):
 			raise KeyError(cpv)
-		mytbz2 = portage.xpak.tbz2(tbz2path)
-		mydata = mytbz2.get_data()
+
+		if binpkg_path.endswith(SUPPORTED_XPAK_EXTENSIONS):
+			mytbz2 = portage.xpak.tbz2(binpkg_path)
+			mydata = mytbz2.get_data()
+			encoding_key = True
+		elif binpkg_path.endswith(SUPPORTED_GPKG_EXTENSIONS):
+			mybinpkg = portage.gpkg.gpkg(self.settings, cpv, binpkg_path)
+			mydata = mybinpkg.get_metadata()
+			encoding_key = False
+		else:
+			return
```

We should raise InvalidBinaryPackageFormat here.